numpy/fft/pocketfft/pocketfft.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * This file is part of pocketfft.
+ * Licensed under a 3-clause BSD style license - see LICENSE.md
+ */
+
+/*! \file pocketfft.h
+ *  Public interface of the pocketfft library
+ *
+ *  Copyright (C) 2008-2018 Max-Planck-Society
+ *  \author Martin Reinecke
+ */
+
+#ifndef POCKETFFT_H
+#define POCKETFFT_H
+
+#include <stdlib.h>
+
+struct cfft_plan_i;
+typedef struct cfft_plan_i * cfft_plan;
+cfft_plan make_cfft_plan (size_t length);
+void destroy_cfft_plan (cfft_plan plan);
+int cfft_backward(cfft_plan plan, double c[], double fct);
+int cfft_forward(cfft_plan plan, double c[], double fct);
+size_t cfft_length(cfft_plan plan);
+
+struct rfft_plan_i;
+typedef struct rfft_plan_i * rfft_plan;
+rfft_plan make_rfft_plan (size_t length);
+void destroy_rfft_plan (rfft_plan plan);
+int rfft_backward(rfft_plan plan, double c[], double fct);
+int rfft_forward(rfft_plan plan, double c[], double fct);
+size_t rfft_length(rfft_plan plan);
```

Comment:
My logic was that we're vendoring the pocketfft files, so ideally we really change as little as possible. Does it matter if it is only included in the ufunc generation code? If you think it does matter, even a little, then I'll change it. But unlike the other changes to pocketfft, we could not upstream these.

Reply:
I am not super sure what the advantags are beyond making them more private (i.e. if it can avoid clashs or has other advantages).

So all else being equal, it is the right thing to add it I think.  But it probably doesn't matter a whole lot, so :shrug:.

Reply:
OK, I just added `NPY_VISIBILITY_HIDDEN` (which is the same as `NPY_NO_EXPORT`, but defined in `numpy/numpyconfig.h` rather than somewhere in `multiarray`, so a bit more logical for a vendored routine).